client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -12,9 +12,67 @@ import { translate } from 'i18n-calypso';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { errorNotice } from 'state/notices/actions';
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+import {
+	JETPACK_ONBOARDING_SETTINGS_REQUEST,
+	JETPACK_ONBOARDING_SETTINGS_SAVE,
+} from 'state/action-types';
+import { getUnconnectedSite } from 'state/selectors';
 import { updateJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
 
+export const announceRequestFailure = ( { dispatch } ) =>
+	dispatch(
+		errorNotice( translate( 'Could not fetch settings from site. Please try again later.' ) )
```

Comment:
I think this message would be confusing to users, so we probably want to be a little more passive in that case. Also, we don't provide an option for the user to retry in an easy way right now.

I'd expect one of the following things to happen in case we're unable to fetch the settings:

* Fall back to empty fields/controls (like the JPO flow currently is) - this makes sense, because JPO will be used mostly by users who have just installed their WordPress, so they're starting from scratch.
* After a couple of retry attempts, redirect back to wp-admin, because there's likely a communication error and we don't want to ruin the user's experience by providing a non-working onboarding flow.

Reply:
I think I'm going for empty fields for now :slightly_smiling_face:

Reply:
Sounds well 👍

Reply:
Added a note in the project board for gracefully handling communication errors, would be nice to have.

Reply:
Failing silently for now per d3874ed.